File: problem_036.py
#!/usr/bin/env python

print("Project Euler - Problem 36:")
print("Find the sum of all numbers, less than one million, which are palindromic in base 10 and base 2.\n")

sum = 0
for n in range(1000000):
    bn = bin(n)[2:]
    if bn == str(bn)[::-1]:
        if n == int(str(n)[::-1]):
            sum += n

print(sum)

File: problem_036.js
#!/usr/bin/node

console.log("Project Euler - Problem 36:");
console.log("Find the sum of all numbers, less than one million, which are palindromic in base 10 and base 2.\n");

function is_palindromic(n){
    if (n == n.split("").reverse().join(""))
        return true;
    return false;
}

const limit = 1000000;
var sum = 0;

for (var n=1; n<limit; n++)
    if (is_palindromic(Number(n).toString()) && is_palindromic(Number(n).toString(2)))
        sum += n;

console.log(sum);
